### PR TITLE
Support Rails 7.2

### PR DESCRIPTION
**Pull request by olivier-thatch**
Support Rails 7.2

[Rails 7.2](https://github.com/rails/rails/releases/tag/v7.2.0) was released a few weeks ago. This PR upgrades minitest-rails accordingly.

Note that Rails 7.2 bumped the minimum Ruby version from 2.7 to 3.1, so I've done the same here.

### CODE REVIEW

**Comment by olivier-thatch** [comment]
@blowmage Any chance this could be released soon?

**Comment by zenspider** [comment]
@blowmage nudge

**Comment by n-rodriguez** [comment]
Hi there! Any news? ping @blowmage

**Comment by bquorning** [comment]
Based off of this branch, I opened #258 to add support for Rails 8.0 too.

File: .rubocop.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 AllCops:
-  TargetRubyVersion: 2.7
+  TargetRubyVersion: 3.1
```
Comment (zenspider) [inline .rubocop.yml:2]: Please no... this isn't a part of supporting rails 7.2 or is it necessary for it, so it should be a separate PR.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -18,19 +18,11 @@ jobs:
...
           - '3.2'
           - '3.3'
         gemfile:
-          - gemfiles/rails-7.1.0.gemfile
-          - gemfiles/rails-7.2.0.beta2.gemfile
-        exclude:
-          - ruby: '2.7'
-            gemfile: gemfiles/rails-7.2.0.beta2.gemfile
-          - ruby: '3.0'
-            gemfile: gemfiles/rails-7.2.0.beta2.gemfile
```
Comment (zenspider) [inline .github/workflows/ci.yml:33]: I'd argue that we can't drop as many ruby versions because we're beholden to what rails supports.

File: minitest-rails.gemspec [hunk 3]
```diff
@@ -15,10 +15,10 @@ Gem::Specification.new do |gem|
                       ["README.md", "UPDATING.md", "CHANGELOG.md", "LICENSE"]
   gem.require_paths = ["lib"]
 
-  gem.required_ruby_version = ">= 2.7.0"
+  gem.required_ruby_version = ">= 3.1.0"
```
Comment (zenspider) [inline minitest-rails.gemspec:18]: also no

**Review by zenspider — CHANGES_REQUESTED** [review]
OK. I'mma step in and try to maintain this for a while... As such I'm still catching up... I'd like to request some changes to strip this down to JUST supporting rails 7.2. The rest of this PR should be broken out to a separate commit and PR please.

**Comment by zenspider** [comment]
Also I understand if it's been too long... I'm happy to make the changes requested.

File: minitest-rails.gemspec (hunk 3 above)
Comment (olivier-thatch) [reply minitest-rails.gemspec:18]: I'll revert the changes as asked, but there is still a dependency on railties >= 7.2.0 which itself depends on Ruby >= 3.1 ([code](https://github.com/rails/rails/blob/v7.2.0/railties/railties.gemspec#L12)).

**Comment by olivier-thatch** [comment]
@zenspider Made the requested changes.

**Comment by n-rodriguez** [comment]
Hi there! Any news?

**Comment by zenspider** [comment]
I'm on it

**Review by zenspider — APPROVED** [review]

**Comment by zenspider** [comment]
All the 7.2 tests passed and all the 7.1 tests failed, presumably because they have too old of a ruby version? I am trying to make sense of this but since the 7.2 tests are passing I'mma consider this good and I'll fix the CI out of band.
